src/api/posts.ts

```diff
@@ -0,0 +1,25 @@
+import { Post } from '../react-app-env';
+
+export const BASE_URL = 'https://mate.academy/students-api/';
+
+export async function getPosts(userId: string): Promise<Post[]> {
```

Comment:
It does not make sense to duplicate the logic of the function of sending a request to receive posts. You can rewrite the getPosts function for cases where the userId parameter is not passed.
```suggestion
export async function getPosts(userId?: string): Promise<Post[]> {
  const params = new URLSearchParams(userId ? { userId } : {});
  const response = await fetch(`${BASE_URL}/posts?${params}`);
  if (!response.ok) {
    throw new Error(`${response.status} - ${response.statusText}`);
  }
  return response.json();
}
```